source/nodejs/adaptivecards/src/card-elements.ts

```diff
@@ -4638,6 +4656,7 @@ export abstract class Action extends CardObject {
     private _state: ActionButtonState = ActionButtonState.Normal;
     private _actionCollection?: ActionCollection; // hold the reference to its action collection
     private _isFocusable: boolean = true;
+    private _isDesignMode: boolean;
```

Comment:
designer and adaptive cards framework should be independent of each other.
it appears that adaptive cards is aware of designer.

Reply:
Yes, `card-elements.ts` already has `isDesignMode()` which is what we use to assign this value.

https://github.com/microsoft/AdaptiveCards/blob/7d91f67fb8707231c8fb677e5cc8f50df79cdc75/source/nodejs/adaptivecards/src/card-elements.ts#L596-L600

The `rootElement.designMode` defaults to `false`